domain/operation/state/types.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package state
+
+import (
+	"database/sql"
+	"time"
+)
+
+// operationRow represents the struct for querying the operation table.
+type operationRow struct {
+	UUID           string     `db:"uuid"`
+	Summary        *string    `db:"summary"`
+	EnqueuedAt     time.Time  `db:"enqueued_at"`
+	StartedAt      *time.Time `db:"started_at"`
+	CompletedAt    *time.Time `db:"completed_at"`
+	Parallel       bool       `db:"parallel"`
+	ExecutionGroup *string    `db:"execution_group"`
+}
+
+// operationTaskRow represents the struct for querying the operation_task table.
+type operationTaskRow struct {
+	UUID          string     `db:"uuid"`
+	OperationUUID string     `db:"operation_uuid"`
+	EnqueuedAt    time.Time  `db:"enqueued_at"`
+	StartedAt     *time.Time `db:"started_at"`
+	CompletedAt   *time.Time `db:"completed_at"`
+}
+
+// operationUnitTaskRow represents the struct for querying operation_unit_task
+// table.
+type operationUnitTaskRow struct {
+	TaskUUID string `db:"task_uuid"`
+	UnitUUID string `db:"unit_uuid"`
+}
+
+// operationMachineTaskRow represents the struct for querying
+// operation_machine_task table.
+type operationMachineTaskRow struct {
+	TaskUUID    string `db:"task_uuid"`
+	MachineUUID string `db:"machine_uuid"`
+}
+
+// operationTaskStatusRow represents the struct for querying
+// operation_task_status table.
+type operationTaskStatusRow struct {
+	TaskUUID  string     `db:"task_uuid"`
+	StatusID  int        `db:"status_id"`
```

Comment:
todo: The db is the source of truth for the mapping of status values to status ids.

`Status corestatus.Status `db:"status"`